src/System.Reflection.Metadata/src/System/Reflection/Metadata/Internal/MetadataFlags.cs

```diff
@@ -120,86 +122,194 @@ internal enum HeapSizeFlag : byte
 
     internal enum StringKind : byte
     {
-        Plain = 0,
-        WinRTPrefixed = 1,
-        DotTerminated = 2,
+        Plain = (byte)(HeapHandleType.String >> HeapHandleType.OffsetBitCount),
+        Virtual = (byte)(HeapHandleType.VirtualString >> HeapHandleType.OffsetBitCount),
+        WinRTPrefixed = (byte)(HeapHandleType.WinRTPrefixedString >> HeapHandleType.OffsetBitCount),
+        DotTerminated = (byte)(HeapHandleType.DotTerminatedString >> HeapHandleType.OffsetBitCount),
     }
 
     internal enum NamespaceKind : byte
     {
-        Plain = 0,
-        Synthetic = 1,
+        Plain = (byte)(HeapHandleType.Namespace >> HeapHandleType.OffsetBitCount),
+        Synthetic = (byte)(HeapHandleType.SyntheticNamespace >> HeapHandleType.OffsetBitCount),
     }
 
-    internal static class TokenTypeIds
+    internal static class HeapHandleType
```

Comment:
This name is confusing. From it, I'm surprised that HeapHandleType.Namespace == HeapHandleType.String. This is just about strings and namespaces and not generally about heap handles (e.g. not blobs, guids). Maybe split out String and Namespace and include it in class name?